superset/assets/javascripts/explore/components/ChartContainer.jsx

```diff
@@ -152,15 +152,20 @@ class ChartContainer extends React.PureComponent {
     this.props.actions.runQuery(this.props.formData, true, this.props.timeout);
   }
 
-  updateChartTitle(newTitle) {
+  updateChartTitleOrSaveSlice(newTitle) {
+    const isNewSlice = !this.props.slice;
     const params = {
       slice_name: newTitle,
-      action: 'overwrite',
+      action: isNewSlice ? 'saveas' : 'overwrite',
     };
     const saveUrl = getExploreUrl(this.props.formData, 'base', false, null, params);
     this.props.actions.saveSlice(saveUrl)
-      .then(() => {
-        this.props.actions.updateChartTitle(newTitle);
+      .then((data) => {
+        if (isNewSlice) {
+          window.location = data;
```

Comment:
Is a round trip necessary? As a user it may be confusing.

Reply:
after new slice is created, we render "favorite" and "edit slice properties" icon on top of the slice. For chart in visualize flow we don't show those icons.
Refresh page after save slice_name is just to render those icons.